src/bot_methods.js

```diff
@@ -410,29 +296,44 @@ function handleCommandHelp(message) {
   message.channel.send(help);
 }
 
+/**
+ * Handles the 'hitme' command
+ * continuously provides words to translate
+ * @param {Message} message
+ */
 function handleChallengeMode(message) {
-  let newWord = challengeMode(message);
-  console.log(newWord[0]);
-  console.log(newWord[1]);
-  const filter = response => response.content.toLowerCase() == newWord[1].toLowerCase() || response.content.toLowerCase() == 'stop' || response.content.toLowerCase() == 'next';
-  message.channel.send("Translate: " + newWord[0]).then(() => {
-    message.channel.awaitMessages(filter, { maxMatches: 1, time: 15000, errors: ['time'] })
-      .then(collected => {
-        console.log(collected.first().content);
-        if (collected.first().content.toLowerCase() == 'stop') {
+  let wordCheck = false;
+  let newWord = null;
+  while (!wordCheck){ // because some words lack english translations, it loops until a word with a translation is found
+    let randWord = getRandomWord(message); // placeholder
+    newWord = [randWord["word"], randWord["english"]]; // a word and translation
+    if(newWord[1] != null && newWord[1] !== "") {
+      wordCheck = true;
+    }
+  }
+  console.log(newWord[0] + ": " + newWord[1]);
+  const filter = response => response.content.toLowerCase() == newWord[1].toLowerCase() // the filter for accepting user input.
+    || response.content.toLowerCase() == STOP  // if the user types stop, next or the correct word,
+    || response.content.toLowerCase() == NEXT; // it will return true. otherwise false
+  message.channel.send("Translate: **" + newWord[0] + "**. (type " + STOP + "to end)").then(() => {
+    message.channel.awaitMessages(filter, { maxMatches: 1, time: 15000, errors: ['time'] }) // new message collector which accepts one match, in a certain time limit ('time')
+      .then(collected => { // this only runs if the filter returns true
+        let input = collected.first().content.toLowerCase() ;
+        console.log("Recieved word: " + input);
+        if (input == STOP) { // if the user entered stop
           message.channel.send('All done!');
-          return;
-        } else if (collected.first().content.toLowerCase() == 'next') {
-          message.channel.send('The answer is: **' + newWord[1] + '**');
+          return; // ends the message collector
+        } else if (input == NEXT) { // if the user entered 'next'

```

Comment:
I forgot to mention, in js its best practice to use `===` instead of `==`, (and `!==` vs `!=`)
because it also checks for variable type.

Reply:
alright - wasn't aware of this operator